_posts/2024-05-14-registry-dereference.md

```diff
@@ -0,0 +1,285 @@
+
+    // Send/produce the message on the Kafka Producer
+    ProducerRecord<Object, Object> producedRecord = new ProducerRecord<>(topicName, SUBJECT_NAME, citizen);
+    producer.send(producedRecord);
+}
+```
+
+And consume them with a consumer that is created as follows, instructing the deserializer to use the schema dereferenced with the following configuration property _DESERIALIZER_DEREFERENCE_SCHEMA_:
+
+```java
+Properties props = new Properties();
+
+props.putIfAbsent(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, SERVERS);
+props.putIfAbsent(ConsumerConfig.GROUP_ID_CONFIG, "Consumer-" + TOPIC_NAME);
+props.putIfAbsent(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
+props.putIfAbsent(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, "1000");
+props.putIfAbsent(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+props.putIfAbsent(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+props.putIfAbsent(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, JsonSchemaKafkaDeserializer.class.getName());
+props.putIfAbsent(SerdeConfig.ARTIFACT_RESOLVER_STRATEGY, SimpleTopicIdStrategy.class.getName());
+props.putIfAbsent(SerdeConfig.EXPLICIT_ARTIFACT_GROUP_ID, "default");
+props.putIfAbsent(SerdeConfig.VALIDATION_ENABLED, true);
+
+//This is the most important configuration for this example, that instructs the deserializer to fetch the dereferenced schema.
+props.putIfAbsent(SchemaResolverConfig.DESERIALIZER_DEREFERENCE_SCHEMA, "true");
+
+// Configure Service Registry location
+props.putIfAbsent(SerdeConfig.REGISTRY_URL, REGISTRY_URL);
+// No other configuration needed for the deserializer, because the globalId of the schema
+// the deserializer should use is sent as part of the payload.  So the deserializer simply
+// extracts that globalId and uses it to look up the Schema from the registry.
+
+// Create the Kafka Consumer
+KafkaConsumer<Long, Citizen> consumer = new KafkaConsumer<>(props);
+```
+
+And then consume the messages with the consumer configured above like this:
+
+```java
+while (messageCount < 5) {
+    final ConsumerRecords<Long, Citizen> records = consumer.poll(Duration.ofSeconds(1));
+    messageCount += records.count();
+    if (records.count() == 0) {
+        // Do nothing - no messages waiting.
+        System.out.println("No messages waiting...");
+    } else records.forEach(record -> {
+        Citizen msg = record.value();
+        System.out.println("Consumed a message: " + msg + " @ " + msg.getCity());
+    });
+}
+```
+
+
+With all of this, instead of doing an HTTP request to get the _citizen schema_, another one to fetch the pointers to the references, and, in this case (this process grows with the hierarchy size), another nine HTTP requests to discover the full hierarchy, a single HTTP request is made that gets the full content of the hierarchy and then uses it to produce/consume messages.
```

Comment:
```suggestion
Instead of doing an HTTP request to get the _citizen schema_, another request to fetch the pointers to the references, and, in this case (this process grows with the hierarchy size), another nine requests to discover the full hierarchy, a single HTTP request gets the full content of the hierarchy and then uses it to produce/consume messages.
```